pkg/virt-controller/watch/vmi.go

```diff
@@ -570,6 +639,9 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 				delete(vmiCopy.Labels, virtv1.OutdatedLauncherImageLabel)
 			}
 			vmiCopy.Status.LauncherContainerImageVersion = ""
+			if util.IsAutoAttachVsock(vmiCopy) {
+				c.cidsMap.Remove(vmiCopy)
```

Comment:
If someone deletes and forcefully removes the finalizers, we may not reach this point. How about removing the CID from the map when the controller sees that the VMI is really gone? 

In `execute()` there is

```go
	if !exists {
		c.podExpectations.DeleteExpectations(key)
		c.vmiExpectations.DeleteExpectations(key)
		return nil
}
```

How about removing the CID there? This way a VMI would keep the CID, even if finalized, until it is really gone, also ensuring that no new VMI could get the same CID assigned.

Reply:
I thought about this. The only thing is that we don't have the VMI object here. We need to use the same key function in the CID map. However, in this file `controller.VirtualMachineInstanceKey` and `controller.KeyFunc` are both used. Although they may produces the same value, I think I should use `controller.KeyFunc`, correct?

Reply:
Yes, you directly need to use the key. That sounds ok to me.